activemodel/lib/active_model/type/model.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module ActiveModel
+  module Type
+    class Model < Value # :nodoc:
+      def initialize(**args)
+        @class_name = args.delete(:class_name)
```

Comment:
What do you think about supporting both `:class` and `:class_name` options?

```suggestion
        @class = args.delete(:class)
        @class_name = args.delete(:class_name)
```

Then, `#klass` could be implemented like:

```ruby
def klass
  @_model_type_class ||= @class || @class_name.constantize
end
```

Reply:
No. Accepting a class reference means that classes will be eager loaded by defining attributes. This can create circular loading, deadlocks, and loading classes too early in the boot process.